models/quix/optimism/quix_v2_optimism_events.sql

```diff
@@ -0,0 +1,237 @@
+    )
+}}
+{% set quix_fee_address_address = "0xec1557a67d4980c948cd473075293204f4d280fd" %}
+{% set min_block_number = 2753613 %}
+
+
+with events_raw as (
+    select 
+      *
+    from (
+        select
+            evt_block_number as block_number
+            ,tokenId as token_id
+            ,contract_address as project_contract_address
+            ,evt_tx_hash as tx_hash
+            ,evt_block_time as block_time
+            ,buyer
+            ,seller
+            ,erc721address as nft_contract_address
+            ,price as amount_raw
+        from {{ source('quixotic_v2_optimism','ExchangeV2_evt_BuyOrderFilled') }}
+        {% if is_incremental() %} -- this filter will only be applied on an incremental run
+        where evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+
+        union all 
+
+        select
+            evt_block_number as block_number
+            ,tokenId as token_id
+            ,contract_address as project_contract_address
+            ,evt_tx_hash as tx_hash
+            ,evt_block_time as block_time
+            ,buyer
+            ,seller
+            ,erc721address as nft_contract_address
+            ,price as amount_raw
+        from {{ source('quixotic_v2_optimism','ExchangeV2_evt_DutchAuctionFilled') }}
+        {% if is_incremental() %} -- this filter will only be applied on an incremental run
+        where evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+
+        union all 
+
+        select
+            evt_block_number as block_number
+            ,tokenId as token_id
+            ,contract_address as project_contract_address
+            ,evt_tx_hash as tx_hash
+            ,evt_block_time as block_time
+            ,buyer
+            ,seller
+            ,erc721address as nft_contract_address
+            ,price as amount_raw
+        from {{ source('quixotic_v2_optimism','ExchangeV2_evt_SellOrderFilled') }}
+        {% if is_incremental() %} -- this filter will only be applied on an incremental run
+        where evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+    ) as x 
+    where nft_contract_address != lower('0xbe81eabdbd437cba43e4c1c330c63022772c2520') -- --exploit contract
+)
+,transfers as (
+    -- eth royalities
+    select 
+      tr.tx_block_number as block_number
+      ,tr.tx_block_time as block_time
+      ,tr.tx_hash 
+      ,tr.value
+      ,tr.to
+    from events_raw as er 
+    join {{ ref('transfers_optimism_eth') }} as tr 
+      on er.tx_hash = tr.tx_hash 
+      and er.block_number = tr.tx_block_number
+      and tr.value_decimal > 0
+      and tr.to not in (
+        lower('{{quix_fee_address_address}}') --qx platform fee address
+        ,er.seller
+        ,er.project_contract_address
+      )
+      {% if not is_incremental() %}
+      -- smallest block number for source tables above
+      and tr.tx_block_number > '{{min_block_number}}'
```

Comment:
does this need to be ```>=``` to include the min block number?

Reply:
i made the min block as actual min in the table - 1... or maybe that's just confusing?

Reply:
maybe a slight bit of confusion, but that's okay 🙏